var/spack/repos/builtin/packages/xyce/package.py

```diff
@@ -37,6 +38,8 @@ class Xyce(CMakePackage):
     variant('mpi', default=True, description='Enable MPI support')
     depends_on('mpi', when='+mpi')
 
+    variant('shared', default=False, description='Enable shared libraries for Xyce')
```

Comment:
I see that the previous script had code to check for `+shared` but the variant was missing. This might be a good opportunity to write:
```python
        if '+shared' in spec:
            options.append('-DBUILD_SHARED_LIBS:BOOL=ON')
        else:
            options.append('-DBUILD_SHARED_LIBS:BOOL=OFF')
```
as
```python
    options.append(self.define_from_variant('BUILD_SHARED_LIBS', 'shared'))
```

Reply:
Great suggestion. Thanks!